3_transformer.go

```diff
@@ -81,7 +82,10 @@ func generateCommitHashURL(remoteURL *url.URL, longHash string) string {
 	}
 
 	if remoteURL != nil {
-		u, _ := url.Parse(remoteURL.String())
+		u, err := url.Parse(remoteURL.String())
+		if err != nil {
+			panic(fmt.Errorf("could not re-parse remote url, check git remote: %w", err))
```

Comment:
I think this should return an error may be better:

```go
return nil, errors.WithStack(err)
```

Reply:
I am going to try and copy the struct instead of re-parsing the string, which looks to be safe to do according to https://github.com/golang/go/issues/38351